pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -598,8 +598,190 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
-				ret = append(ret, &junitapi.JUnitTestCase{Name: testName})
+				ret = append(ret, &junitapi.JUnitTestCase{Name: testName, SystemOut: "Passing the case to make the overall test case flake as the previous failure is expected"})
+			}
+		}
+	}
+
+	return ret
+}
+
+func clusterOperatorIsNotProgressingWhenMachineConfigIs(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	var ret []*junitapi.JUnitTestCase
+	upgradeWindows := getUpgradeWindows(events)
+
+	var machineConfigProgressingStart time.Time
+	var eventsInUpgradeWindows monitorapi.Intervals
+
+	var start, stop time.Time
+	for _, event := range events {
+		if !isInUpgradeWindow(upgradeWindows, event) {
+			continue
+		}
+		eventsInUpgradeWindows = append(eventsInUpgradeWindows, event)
+		if start.IsZero() || event.From.Before(start) {
+			start = event.From
+		}
+		if stop.IsZero() || event.To.After(stop) {
+			stop = event.To
+		}
+	}
+	duration := stop.Sub(start).Seconds()
+
+	eventsByOperator := getEventsByOperator(eventsInUpgradeWindows)
+	for _, mcEvent := range eventsByOperator["machine-config"] {
+		condition := monitorapi.GetOperatorConditionStatus(mcEvent)
+		if condition == nil {
+			continue // ignore non-condition intervals
+		}
+		if condition.Type == configv1.OperatorProgressing && condition.Status == configv1.ConditionTrue {
+			machineConfigProgressingStart = mcEvent.To
+			break
+		}
+	}
+
+	mcTestCase := &junitapi.JUnitTestCase{
+		Name:     fmt.Sprintf("[bz-Machine Config Operator] clusteroperator/machine-config must go Progressing=True during an upgrade test"),
+		Duration: duration,
+	}
+	if machineConfigProgressingStart.IsZero() {
+		mcTestCase.FailureOutput = &junitapi.FailureOutput{
+			Output: fmt.Sprintf("machine-config was never Progressing=True during the upgrade window from %s to %s", start.Format(time.RFC3339), stop.Format(time.RFC3339)),
+		}
+		return []*junitapi.JUnitTestCase{mcTestCase}
+	} else {
+		mcTestCase.SystemOut = fmt.Sprintf("machine-config became Progressing=True at %s during the upgrade window from %s to %s", machineConfigProgressingStart.Format(time.RFC3339), start.Format(time.RFC3339), stop.Format(time.RFC3339))
+	}
+	ret = append(ret, mcTestCase)
+
+	for _, operatorName := range platformidentification.KnownOperators.Difference(sets.NewString("machine-config")).List() {
+		bzComponent := platformidentification.GetBugzillaComponentForOperator(operatorName)
+		testName := fmt.Sprintf("[bz-%v] clusteroperator/%v should stay Progressing=False while MCO is Progressing=True", bzComponent, operatorName)
+		operatorEvents := eventsByOperator[operatorName]
+		if len(operatorEvents) == 0 {
+			ret = append(ret, &junitapi.JUnitTestCase{
+				Name:     testName,
+				Duration: duration,
+			})
+			continue
+		}
+
+		except := func(co string, reason string) string {
+			switch co {
+			case "csi-snapshot-controller":
+				if reason == "CSISnapshotController_Deploying" {
+					return "https://issues.redhat.com/browse/OCPBUGS-62624"
```

Comment:
This bug is already in POST state. Who will monitor and remove those exceptions?